bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepositoryCloner.java

```diff
@@ -0,0 +1,25 @@
+package org.jboss.da.bc.backend.api;
+
+import org.jboss.da.scm.SCMType;
+
+/**
+ * 
+ * @author Jakub Bartecek <jbartece@redhat.com>
+ *
+ */
+public interface RepositoryCloner {
+
+    /**
+     * Clones the repository to the internal systems. Source repository has to be available without authentication
+     * 
+     * @param url URL to the source repository
+     * @param revision Revision, which will be cloned
+     * @param scmType Type of the repository. Currently is supported only GIT
+     * @param repositoryName Name of the new repository
+     * @return URL to the new repository
+     * @throws Exception Thrown if the cloning process failed
+     */
+    String cloneRepository(String url, String revision, SCMType scmType, String repositoryName)
```

Comment:
From the recent conversation I had with Alex S and how repour does cloning, is revision a tag, commit id, or can be both?